hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java

```diff
@@ -93,7 +94,8 @@ protected boolean canEvictRecord(Record existingRecord, ExpiredKey expiredKey) {
         // we send it to backups a new record is added with same key, when we send queued item
         // to backups, backups should not remove it. Comparing creation times to be sure that
         // we are deleting correct record.
-        return existingRecord.getCreationTime() == expiredKey.getCreationTime();
+        // since 3.11, creationTime is maintained at second accuracy
+        return existingRecord.getCreationTime() == zeroOutMillis(expiredKey.getCreationTime());
```

Comment:
Isn't there a case where we replicate records with milliseconds accuracy `creationTime` from 3.10 member to a 3.11 member during RU? It seems the equality check doesn't pass that case and we again won't evict. Shouldn't the comparison take the accuracy of the record's `creationTime` into account? If it's seconds only than we should zero out the millis here. Am I missing something?

Reply:
Ok, we zero out the milliseconds in `AbstractRecord` in this case. Ignore me :smile: